indicaAi/app/controllers/favorite_locals_controller.rb

```diff
@@ -0,0 +1,57 @@
+# Favorite Controller
+class FavoriteLocalsController < ApplicationController
+  # GET /favorites
+  def index
+    @favorites = FavoriteLocal.all
+    json_response(@favorites)
+  end
+
+  # Post /favorite/create
+  def create
+    @favorite = FavoriteLocal.new(favorite_params)
+    if @favorite.valid?
+      @favorite.save
+      render_success('SUCCESS', 'Saved favorite', @favorite)
+    else
+      render_error('ERROR', 'Favorite not saved')
+    end
+  end
+
+  # Patch /favorites/update/:id
+  def update
+    @favorite = FavoriteLocal.find(params[:id])
+    @favorite.update(favorite_params)
+
+    if @favorite.save
+      render_success('SUCCESS', 'Updated favorite', @favorite)
+    else
+      render_error('ERROR', 'Not updated favorite')
+    end
+  end
+
+  # Delete /favorites/destroy/:id
+  def destroy
+    @favorite = FavoriteLocal.find(params[:id])
```

Comment:
Antes de executar o return é bom adicionar uma verificação se o @favorite realmente existe, porque essa linha pode retornar nil.

Reply:
404 pode ser porque eu to chamando a url `favorite/destroy/:id/` ou `favorites/destro/:id` ou esse caso da id que não existe no sistema. Mas é um erro que pode acontecer por N motivos. Você pode retornar o status http 404, mas se você retorna junto uma mensagem que diz que a id não foi encontrada no banco de dados você facilita pro desenvolvedor do front a identificar o problema, e até a forma como ele vai tratar esse erro no front na hora de exibir pro usuário.